docker/Dockerfile

```diff
@@ -49,25 +49,33 @@ ENV PYTHONUNBUFFERED=1 \
     NAUTOBOT_ROOT=/opt/nautobot \
     prometheus_multiproc_dir=/prom_cache
 
+ARG NODE_MAJOR=18
+
 # DL3008 - Require pinned dependencies for apt install
 # DL3009 - Delete the apt-get lists after installing something
-# DL3013 - pin all Python package versions
-# DL3042 - run pip install with --no-cache-dir (https://github.com/hadolint/hadolint/issues/497)
 # DL4006 - Set the SHELL option -o pipefail before RUN with a pipe in
-# hadolint ignore=DL3008,DL3009,DL3013,DL3042,DL4006
-RUN --mount=type=cache,target="/root/.cache/pip",sharing=locked \
+# hadolint ignore=DL3008,DL3009,DL4006
+RUN --mount=type=cache,target="/var/cache/apt",sharing=locked \
+    --mount=type=cache,target="/var/lib/apt/lists",sharing=locked \
+    apt-get update && \
+    apt-get install --no-install-recommends -y ca-certificates curl gnupg && \
+    mkdir -p /etc/apt/keyrings && \
+    curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg && \
+    echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list && \
     apt-get update && \
     apt-get upgrade -y && \
-    apt-get install --no-install-recommends -y curl && \
-    curl -fsSL https://deb.nodesource.com/setup_18.x | bash - && \
     apt-get install --no-install-recommends -y git mime-support libxml2 libmariadb3 openssl nodejs && \
-    apt-get autoremove -y && \
-    apt-get clean all && \
```

Comment:
Is this line not required?

Reply:
> Is this line not required?

It's not necessary, as apt temporary folders are mounted as cache, lines 58, 59

Reply:
> Same question for the rest of the cleanup lines removed below

The same, solved by mounting apt temporary folders to Docker cache.

Reply:
> We need the `rm -rf /tmp/tmp*`. It tries to clean up that directory as the nautobot user later in the `python-dependencies` stage and fails

`/mnt` is mounted as a Docker cache at line 77, so the line 78: `RUN pip install ...` instruction do not leave any files in /tmp on the image.

Reply:
This build was failing for me when trying to delete a `/tmp/tmp<random string>` file yesterday but seems to be working now. Maybe I had a bad cache.